test/lib/clients/inst_productsources_test.rb

```diff
@@ -46,6 +49,16 @@
 
       client.main
     end
+
+    it "sets the RELEASEVER_ENV to Product.version" do
+      stub_const("#{described_class}::RELEASEVER_ENV", "ZYPP_RELEASEVER_ENV")
+
+      expect(ENV[described_class::RELEASEVER_ENV]).to be_nil
+
+      client.main
+
+      expect(ENV[described_class::RELEASEVER_ENV]).to eq(product_version)
```

Comment:
This test actually really modifies the environment... :worried: 

What about stubbing the change like this:

```ruby
expect(ENV).to receive(:[]=).with(...)
```

The modified the ENV could leak to another tests and cause random failures...

Reply:
Intersting, because it was my first idea... but I though in that way I could not test the key, only the value. But I did not test it :sweat_smile: 

Now, I can see that I can do it, simply given two params to `.with` :sweat_smile: :sweat_smile: 

Thanks!